migrator/data/DataImporter.java

```diff
@@ -248,6 +248,7 @@ private void commitBatch() {
                 conceptTracker.recordMapped(originalID, pair.second());
                 if (incompleteIDs.contains(originalID)) conceptTracker.recordIncomplete(originalID);
             });
+            incompleteIDs.clear();
```

Comment:
I could also add an assert to line 244, asserting the incompleteIDs are either in the originalToBufferedIDs set, or in the conceptTracker's incomplete set.